### PR TITLE
Syslog validation

### PR DESCRIPTION
**Pull request by LK4D4**
This PR adds:
- validation of corrupted syslog addreses
- validation of this on daemon start
- Fix for logconfig.Config in inspect
- test for daemon wide log-opt
  ping @cpuguy83

### CODE REVIEW

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -1550,3 +1550,23 @@ func (s *DockerDaemonSuite) TestDaemonRestartWithContainerWithRestartPolicyAlway
 	c.Assert(err, check.IsNil)
 	c.Assert(strings.TrimSpace(out), check.Equals, id[:12])
 }
+
+func (s *DockerDaemonSuite) TestDaemonCorruptedSyslogAddress(c *check.C) {
+	c.Assert(s.d.Start("--log-driver=syslog", "--log-opt", "syslog-address=corrupted:1234"), check.NotNil)
+	runCmd := exec.Command("grep", "Failed to set log opts: syslog-address should be in form proto://address", s.d.LogfileName())
```
Comment (cpuguy83) [inline integration-cli/docker_cli_daemon_test.go]: Would the error returned by `Start` have this output as well?

Comment (LK4D4) [inline integration-cli/docker_cli_daemon_test.go]: Nope :)

**Comment by cpuguy83** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Pull request merged**